spec/tracking_spec.js

```diff
@@ -0,0 +1,92 @@
+describe('App Tracking', function() {
+  var tracking = require('tracking'),
+      sandbox  = sinon.sandbox.create(),
+      client = { invoke: function() {} },
+      eventListener;
+
+  beforeEach(function() {
+    sandbox.stub(client, 'invoke');
+    eventListener = sinon.spy(window, 'addEventListener');
+    tracking.setup(client);
+  });
+
+  afterEach(function() {
+    window.addEventListener.restore();
+    sandbox.restore();
+  });
+
+  function findEvent(event) {
+    for(var i = 0; i < eventListener.args.length; i++) {
+      if(eventListener.args[i][0] === event) {
+        return eventListener.args[i];
+      }
+    }
+  }
+
+  describe('for click events', function() {
+    it('attaches the click event listener', function() {
+      var clickEvent = findEvent('click');
+
+      expect(clickEvent).not.to.be.undefined;
+      expect(clickEvent[0]).to.equal('click');
+      expect(clickEvent[1]).to.be.a.function;
+    });
+  });
+
+  describe('for mouseover events', function() {
+    it('attaches the mouseover event listener', function() {
+      var mouseoverEvent = findEvent('mouseover');
+
+      expect(mouseoverEvent).not.to.be.undefined;
+      expect(mouseoverEvent[0]).to.equal('mouseover');
+      expect(mouseoverEvent[1]).to.be.a.function;
+    });
+
+    it('attaches the mousout event listener', function() {
+      var mouseoutEvent = findEvent('mouseout');
+      
+      expect(mouseoutEvent).not.to.be.undefined;
+      expect(mouseoutEvent[0]).to.equal('mouseout');
+      expect(mouseoutEvent[1]).to.be.a.function;
```

Comment:
Test code looks different to normal code. It's easy to fall out of the good habits that we follow for normal code. Here is the third repeat of these three lines of code:

``` javascript
expect(mouseoutEvent).not.to.be.undefined;
expect(mouseoutEvent[0]).to.equal('mouseout');
expect(mouseoutEvent[1]).to.be.a.function;
```

I suggest pulling it out into a helpfully named function and parameterising it based on the event type. Honestly you could probably pull in

``` javascript
var mouseoutEvent = findEvent('mouseout');
```

as well. Maybe `function expectEventToBeRegistered(eventType) { ... }`.